lua/entities/gmod_tardis/modules/sh_tracking.lua

```diff
@@ -0,0 +1,497 @@
+-- Tracking
+
+local TRACKING_MAX_DISTANCE_SET = 1000
+local TRACKING_MAX_DISTANCE_TARGET_MAX = 1500
+local TRACKING_MAX_DISTANCE_NO_LOS = 2000
+local TRACKING_MAX_DISTANCE_TRACE = 10000
+
+local TRACKING_DEBUG = false
+
+TARDIS:AddKeyBind("tracking",{
+    name="Tracking",
+    section="Flight",
+    func=function(self,down,ply)
+        local pilot = self:GetData("pilot")
+        if down and self:CallHook("CanTrack") == false then return end
+        if ply==pilot then
+            if down then
+                self:SetData("tracking-trace",true)
+            else
+                self:SendMessage("tracking-set", {self:GetData("tracking-ent")})
+            end
+        end
+        if not down then
+            self:SetData("tracking-trace",false)
+            self:SetData("tracking-ent",nil)
+        end
+    end,
+    key=KEY_X,
+    clientonly=true,
+    exterior=true
+})
+
+TARDIS:AddKeyBind("tracking-rotation",{
+    name="TrackRotation",
+    section="Flight",
+    func=function(self,down,ply)
+        local pilot = self:GetData("pilot")
+        if not self:GetTracking() or ply ~= pilot or not down then return end
+        self:ToggleTrackRotation()
+        TARDIS:StatusMessage(ply, "Controls.Tracking.Rotation", self:GetTrackRotation())
+    end,
+    key=KEY_T,
+    serveronly=true,
+    exterior=true
+})
+
+if SERVER then
+    function ENT:GetTracking()
+        return self:GetData("tracking-ent")
+    end
```

Comment:
I will most likely want that to exist on the client
There are many potential, yet not planned, use-cases

Seeing info on the screen about the tracked entity would be pretty dope

Reply:
https://github.com/MattJeanes/TARDIS/pull/969/commits/6bea59f4a5d583fbfd2f63d7553911e575638ea3

![image](https://github.com/MattJeanes/TARDIS/assets/2363642/7032c57d-cb9a-466d-a16f-5f8457d748db)